src/modules/link.cpp

```diff
@@ -1,22 +1,119 @@
 //
 // Link Module
 //
-
+#include <Arduino.h>
+#include <FastCRC.h>
```

Comment:
https://github.com/FrankBoesing/FastCRC
this is supposed to be much faster.

Reply:
I would prefer to avoid any extra external dependencies when possible. I did try running their benchmarking code and while it is ~40% faster than the builtin function, its not a big deal; over the course of a second, we spend 20ms doing the crc (2% of the time: 2.5us per bit with builtin crc, 800 bits per packet, 10 packets per second to be overly generous with the numbers from what I was able to do). Bringing that down to 1.2% isnt really a big deal right now -- but we can look at this solution again if we need to save cycles. For now, please use the `_crc_xmodem_update` from `util/crc16.h` so we can avoid the external dependency

Reply:
I received the mega and tried using utils/ccr16.h. I created a function that works fine in a standalone program using the data packet structure but when using with the controller it had strange behavior and caused serial protocol timeouts. I looked at the code and the functions are inline assembler in crc16.h. I have an option to compile with either. For now I am testing with the fastcrc library. Maybe there is a protothread issue with inline assembler.